api/grpcserver/globalstate_service.go

```diff
@@ -184,13 +185,58 @@ func (s GlobalStateService) AccountDataQuery(_ context.Context, in *pb.AccountDa
 }
 
 // SmesherDataQuery returns historical info on smesher rewards
-func (s GlobalStateService) SmesherDataQuery(context.Context, *pb.SmesherDataQueryRequest) (*pb.SmesherDataQueryResponse, error) {
+func (s GlobalStateService) SmesherDataQuery(_ context.Context, in *pb.SmesherDataQueryRequest) (*pb.SmesherDataQueryResponse, error) {
 	log.Info("GRPC GlobalStateService.SmesherDataQuery")
 
-	// TODO: implement me! We don't currently have a way to read rewards per-smesher.
-	// See https://github.com/spacemeshos/go-spacemesh/issues/2068
+	if in.SmesherId == nil {
+		return nil, status.Errorf(codes.InvalidArgument, "`Id` must be provided")
+	}
+	if in.SmesherId.Id == nil {
+		return nil, status.Errorf(codes.InvalidArgument, "`Id.Id` must be provided")
+	}
+
+	smesherIDBytes := in.SmesherId.Id
+	smesherID := types.BytesToNodeID(smesherIDBytes)
+	dbRewards, err := s.Mesh.GetRewardsBySmesherID(smesherID)
+	res := &pb.SmesherDataQueryResponse{}
+
+	if err != nil {
+		log.Error("unable to fetch projected reward state for smesher %v: %s", smesherID, err)
```

Comment:
please use the `log.With()` syntax here, we can discuss on a call if this is unclear

```suggestion
		log.With().Error("unable to fetch projected reward state for smesher", log.String("smesher", smesherID), log.Error(err))
```